identity-iota/src/did/doc/iota_document.rs

```diff
@@ -113,9 +114,12 @@ impl IotaDocument {
   ///
   /// This must be guaranteed safe by the caller.
   pub unsafe fn from_authentication_unchecked(method: IotaVerificationMethod) -> Self {
+    let verification_method_did: DID = method.id().as_ref().clone();
+
     CoreDocument::builder(Default::default())
       .id(method.controller().clone().into())
-      .authentication(method)
+      .verification_method(method.into())
+      .authentication(MethodRef::Refer(verification_method_did))
```

Comment:
Should we add a unit test to check the structure of the generated document, i.e. that it now has a single verification method which the authentication section references? The current tests use `authentication()` which resolves the reference and doesn't check for a verification method.

Reply:
I've added a test that checks if the verification method with the default fragment `#authentication` is the same as what `document.authentication()` resolves to. `IotaDocument` doesn't expose access to the set of `authentication` methods, and I didn't find a way to get a reference to the `CoreDocument` inside. But I think that test should be good enough.

Reply:
I think one can check that the authentication is a reference as follows:

```rust
    match document.document.authentication().first().unwrap().clone().into_inner() {
      MethodRef::Refer(did) => assert_eq!(did.fragment().unwrap_or_default(), "#authentication"),
      MethodRef::Embed(_) => panic!("authentication method should be a reference"),
    }
```